app/classifier/tasks/dropdown/index.cjsx

```diff
@@ -107,38 +105,48 @@ module.exports = React.createClass
     # as it relies on props.task and props.annotation.
     # clear selections, then check each annotation.value
     # and set the corresponding selected option for each select
-    selectedOptions = []
-    optionsKeys = []
-    @props.annotation.value.map (annotation, i) =>
-      if annotation.option
-        [selected] = @getOptions(i, optionsKeys).filter (option) -> option.value is annotation.value
-        optionsKeys.push selected.value if selected?.value
-        selectedOptions[i] = selected
-      else
-        selectedOptions[i] = {label: annotation.value, value: annotation.value}
-    @setState {selectedOptions}
+    if @props.annotation.value.length?
+      for answer, i in @props.annotation.value
+        @handleOptionsKeys(i, answer.value)
 
   handleFocus: ->
     @menus[0].focus()
 
-  getOptions: (i, optionsKeys) ->
-    optionsKey = if optionsKeys.length then optionsKeys.join(';') else '*'
+  handleOptionsKeys: (i, value) ->
```

Comment:
What does `handleOptionsKeys` do? Can the method name be changed to describe what it does?